include/oneapi/dpl/internal/dynamic_selection_impl/sycl_backend.h

```diff
@@ -89,6 +151,9 @@ class sycl_backend
         for (auto e : v)
         {
             global_rank_.push_back(e);
+            if(e.template has_property<sycl::property::queue::enable_profiling>()){
+                has_enable_profiling = true;
+            }
```

Comment:
The default constructor does not have similar runtime checks and therefore always disables profiling. Is that by design or an oversight?

Reply:
I think we should have three consistent behaviors: (1) has_enable_profiling should be true only if all queues have profiling enabled, currently it is true if one or more has it, (2) the default constructor should try to create queues with profiling enabled (but only if all default devices support profiling), and (3) given the previous 2 points, this single has_enable_profiling bool can be checked in submit instead of querying on the queue. We want this approach consistent across all queues in the universe because the timings will be incompatible and no meaningful comparison can be made if the approach differs between queues,

Reply:
Fixed